### ISSUE TITLE
ImageSource.fromAsset() memory leak on iOS

### ISSUE THREAD

**Issue by VepretskiyD** [NativeScript/NativeScript#9694]
### Issue Description

On iOS my app is crashing because it eats all the memory. I used Xcode profiler to find out what's causing it. The main problem is with `VM: CG raster data` that gets created every time I create an image source from an asset that comes from `nativescript camera-plus`. The problem is that memory allocated by these `raster data` doesn't get released after navigating away.
Basically, every ImageSource.fromAsset call allocates 64Mb of memory for new `VM: CG raster data`
`const imgSource = await ImageSource.fromAsset(data);`
After that I resize an image and save it as a file on local file system.
I tried to call Utils.GC() and navigate away with clearHistory: true. It doesn't help.
![image](https://user-images.githubusercontent.com/17052346/145597173-cd8ac420-ce8b-433d-b3db-e926826e7386.png)
When I asked about this issue on [discord](https://discord.com/channels/603595811204366337/828973466534936637/918524105865691166) @rigor789 [suggested](https://discord.com/channels/603595811204366337/828973466534936637/918528067469733978) that the issue may be caused by `ImageAsset.getImageAsync`:
1. ImageSource.fromAsset: https://github.com/NativeScript/NativeScript/blob/e72980c85bfb1d0696734a4f742c77489b35191a/packages/core/image-source/index.ios.ts#L57-L67
2. it calls asset.getImageAsync: https://github.com/NativeScript/NativeScript/blob/e72980c85bfb1d0696734a4f742c77489b35191a/packages/core/image-asset/index.ios.ts#L33-L62
3.  it calls scaleImage: https://github.com/NativeScript/NativeScript/blob/e72980c85bfb1d0696734a4f742c77489b35191a/packages/core/image-asset/index.ios.ts#L64-L75
4. it uses UIGraphicsGetImageFromCurrentImageContext(); which is known to leak memory if not wrapped in it's own autorelease pool of some sorts: https://stackoverflow.com/a/23875557/2270725

and changing
`const resultImage = this.scaleImage(image, requestedSize);`
to
`const resultImage = image`
should make it clear whether `scaleImage` is a problem or not

He was right. When skipping scaleImage it works as expected, no memory leaks.
I changed
`const imgSource = await ImageSource.fromAsset(data);`
  to```  const imgSource = isIOS
        ? new ImageSource(data.nativeImage)
        : await ImageSource.fromAsset(data); ```

and it starts working as expected

### Reproduction

Install https://www.npmjs.com/package/@nstudio/nativescript-camera-plus
```
  <CameraPlus
      @photoCapturedEvent="onPhotoCapturedEvent"
    />
...
methods: {
  onPhotoCapturedEvent(event) {
    const imgSource = await ImageSource.fromAsset(event.data);  
  }
}
```
Every time you take a photo and create a new imgSource it'll allocate some amount in memory and after taking some amount of photos the app gets terminated by OS

### Environment

```yaml
OS: macOS 11.6.1
CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
Shell: /bin/zsh
node: 14.17.4
npm: 6.14.14
nativescript: 8.1.4

# android
java: 15.0.2
ndk: Not Found
apis: Not Found
build_tools: Not Found
system_images: Not Found

# ios
xcode: 13.1/13A1030d
cocoapods: 1.11.2
python: 2.7.16
python3: 3.9.8
ruby: 2.7.4
platforms: 
  - DriverKit 21.0.1
  - iOS 15.0
  - macOS 12.0
  - tvOS 15.0
  - watchOS 8.0
```

### Dependencies

```json
"dependencies": {
  "@master.technology/permissions": "^2.0.1",
  "@nativescript-community/fonticon": "^3.0.0",
  "@nativescript-community/perms": "^2.1.8",
  "@nativescript-community/ui-collectionview": "^4.0.48",
  "@nativescript/core": "~8.1.1",
  "@nativescript/datetimepicker": "^2.1.9",
  "@nativescript/email": "^2.0.5",
  "@nativescript/imagepicker": "^1.0.5",
  "@nativescript/social-share": "^2.0.4",
  "@nativescript/theme": "~3.0.1",
  "@nstudio/nativescript-camera-plus": "^4.1.0",
  "@triniwiz/nativescript-image-cache-it": "^7.0.12",
  "@triniwiz/nativescript-socketio": "^4.0.2",
  "@triniwiz/nativescript-toasty": "^4.1.3",
  "@vue/devtools": "^5.3.4",
  "axios": "^0.21.1",
  "date-fns": "^2.27.0",
  "lodash": "^4.17.21",
  "nativescript-contacts": "^1.6.4",
  "nativescript-contacts-chooser": "^1.1.0",
  "nativescript-contacts-lite": "^0.2.7",
  "nativescript-imagecropper": "^4.0.1",
  "nativescript-mediafilepicker": "^4.0.2",
  "nativescript-menu": "^1.1.7",
  "nativescript-ui-sidedrawer": "^9.0.3",
  "nativescript-vue": "~2.9.0",
  "nativescript-vue-devtools": "^1.5.1",
  "nativescript-vue-navigator": "^1.2.0",
  "patch-package": "^6.4.7",
  "vue-lodash": "^2.1.2",
  "vuex": "^3.6.2"
},
"devDependencies": {
  "@nativescript/android": "~8.1.1",
  "@nativescript/eslint-plugin": "0.0.4",
  "@nativescript/ios": "~8.1.0",
  "@nativescript/webpack": "^5.0.0-beta.15",
  "@vue/eslint-config-standard": "^5.1.2",
  "eslint": "^7.32.0",
  "eslint-config-airbnb-base": "^14.2.1",
  "eslint-import-resolver-custom-alias": "^1.2.0",
  "eslint-plugin-import": "^2.24.0",
  "eslint-plugin-vue": "^7.16.0",
  "nativescript-vue-template-compiler": "~2.9.0",
  "sass": "^1.32.8"
}
```

### 

- [X] I have searched the [existing issues](https://github.com/NativeScript/NativeScript/issues) as well as [StackOverflow](https://stackoverflow.com/questions/tagged/nativescript) and this has not been posted before
- [X] This is a bug report
- [X] I agree to follow this project's [Code of Conduct](https://github.com/NativeScript/NativeScript/blob/master/tools/notes/CONTRIBUTING.md#coc)

**Comment by edusperoni** [comment]
I wonder if changing that to:
```
 		const pool = NSAutoreleasePool.alloc().init();
 		UIGraphicsBeginImageContextWithOptions(requestedSize, false, scaleFactor); 
 		image.drawInRect(CGRectMake(0, 0, requestedSize.width, requestedSize.height)); 
 		const resultImage = UIGraphicsGetImageFromCurrentImageContext(); 
 		UIGraphicsEndImageContext(); 
 		pool.drain();
```
could solve the issue?

Not sure, since we keep a reference to `resultImage`;

**Comment by NathanWalker** [comment]
@VepretskiyD we have published another 8.2 alpha which addresses this specific case:
```
@nativescript/core@8.2.0-alpha.12
```
Let us know if have a chance to try it - we will plan to close this along with 8.2 release.
